File: chess2.0/server/models/Message.cs
using chess2._0.models.gameRoom;
using Newtonsoft.Json;

namespace chess2._0.models;

public class MessageToClient
{
    [JsonProperty("type")]
    public MessageType Type;
    [JsonProperty("params")]
    public GameRoomDto? Params;
    [JsonProperty("roomId")]
    public string RoomId;

    public MessageToClient(MessageType type, GameRoomDto? messParams, string roomId)
    {
        Type = type;
        Params = messParams;
        RoomId = roomId;
    }
}

public class MessageFromClient
{
    [JsonProperty("type")]
    public MessageType Type;
    [JsonProperty("params")]
    public string Params;
    [JsonProperty("roomId")]
    public string RoomId;
}

public enum MessageType
{
    Create,
    Join,
    Init,
    Start,
    Leave,
    Move,
    NextTurn,
    EndGame,
    ChangeFigure,
    Restart,
    Error,
    GiveUp,
    OfferDraw,
    ConfirmDraw
}

File: client/src/models/Message.ts
import { Cell } from './Cell.ts'
import { FigureColors } from './Figure.ts'

export enum MessageType {
  create,
  join,
  init,
  start,
  leave,
  move,
  nextTurn,
  endGame,
  changeFigure,
  restart,
  error,
}

export enum GameWinner
{
  white,
  black,
  draw
}

export interface MessageParams {
  chessBoardState: Cell[]
  isMyTurn: boolean
  turn: number
  winner: GameWinner | null
  color: FigureColors | null
}

export interface MessageFromServer {
  type: MessageType
  params: null | MessageParams
  roomId: string
}

export interface MessageToServer {
  type: MessageType
  params: string
  roomId: string
}